doc/languages-frameworks/gradle.section.md

```diff
@@ -0,0 +1,137 @@
+
+    mkdir -p $out/opt/
+    cp -r ./build/jextract $out/opt/jextract
+
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    makeWrapper "$out/opt/jextract/bin/jextract" "$out/bin/jextract"
+  '';
+
+  meta = with lib; {
+    description = "A tool which mechanically generates Java bindings from a native library headers";
+    homepage = "https://github.com/openjdk/jextract";
+    license = licenses.gpl2Only;
+    maintainers = with maintainers; [ sharzy ];
+  };
+}
+```
+
+However, most of the time the package does have dependencies. In that
+case, you will have to use the man-in-the-middle proxy.
+
+First, add a dependencies update script:
+
+```nix
+passthru.updateDeps = gradle.updateDeps { inherit pname; };
+```
+
+Then, build the update script via `nix-build -A <pname>.updateDeps` and
+run it via `./result`. It will generate `deps.json` in your package
+directory. You can then import it:
+
+```gradle
+mitmCache = gradle.fetchDeps {
+  inherit pname;
+  data = ./deps.json;
+};
+```
+
+## Update Script {#gradle-update-script}
+
+The update script flow:
+
+- Run the derivation's unpack, patch and configure phases
+- Copy the result to the Nix store, alongside the Gradle flags (if set)
+  and `preBuild`
+- Copy it from the Nix store to an `nsjail` sandbox with networking
+  enabled
+- Run `preBuild` in the nsjail sandbox
+- Run the `preBuild` that was passed as an argument to `updateDeps`
+- Run `gradleCommands`, which defaults to `gradle nixDownloadDeps`
+- Run the `postBuild` that was passed as an argument to `updateDeps`
+
+`updateDeps` takes the following arguments:
+
+- `attrPath` - a way to get to the package attr in nixpkgs, usually this
+  is the package name
+- `pname` - an alias for `attrPath`, for convenience. This should only
+  be used when `attrPath` matches `pname`
+- `code` - allows you to completely override the Nix code used to
+  evaluate the package (only relevant for downstream, non-nixpkgs
+  projects)
+- `attrOverrides` - allows you to override certain attributes of the
+  derivation that will be used for copying the source from
+- `preBuild` / `postBuild` - run these commands before/after fetching
+  dependencies
+- `gradleCommands` - run these commands to fetch the dependencies
+- `depsPath` - path to the dependencies lockfile relative to the
+  package. Defaults to `deps.json`.
+- `availablePackages` - packages that will be available in the nsjail
+  sandbox. By default, no extra packages will be available, which is
+  enough for nearly all packages. For example, the MITM proxy supports
+  intercepting `curl` and `wget` traffic, so if the Gradle build script
+  requires them, you may add them here.
+
+## Environment {#gradle-environment}
+
+The Gradle setup hook accepts the following environment variables:
+
+- `mitmCache` - the MITM proxy cache imported using `gradle.fetchDeps`
+- `mitmCachePort` - the MITM proxy port to be used (defaults to 1337)
+- `gradleFlags` - command-line flags to be used for every Gradle
```

Comment:
Typical environment variables in linux would look like MITM_CACHE, MITM_CACHE_PORT or GRADLE_FLAGS.

I think you mean that those are not interpolated via nix code, right?

Reply:
These are environment variables, i.e. you can set it in a `mkDerivation` call